### PR TITLE
Fix BeitFeatureExtractor postprocessing

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
- Fixes a `BeitFeatureExtractor.post_process_semantic_segmentation()` assertion error when no `target_sizes` argument is provided
- Ensures post_process_semantic_segmentation returns a list of int64 PyTorch tensors
- Adds a test to ensure correct post-processing

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/beit/feature_extraction_beit.py [hunk 1]
```diff
@@ -222,3 +225,44 @@ def __call__(
...
+                target_sizes = target_sizes.numpy()
+
+            resized_maps = []
+            semantic_segmentation = semantic_segmentation.numpy()
+
+            for idx in range(len(semantic_segmentation)):
+                resized = self.resize(image=semantic_segmentation[idx], size=target_sizes[idx])
+                resized_maps.append(resized)
+
+            semantic_segmentation = [torch.Tensor(np.array(image)).to(torch.int64) for image in resized_maps]
```
Comment (NielsRogge) [inline src/transformers/models/beit/feature_extraction_beit.py:264]:
```suggestion
            semantic_segmentation = [torch.Tensor(np.array(image)).long() for image in resized_maps]
```
This should also work :)

File: src/transformers/models/beit/feature_extraction_beit.py [hunk 2]
```diff
@@ -222,3 +225,44 @@ def __call__(
...
+
+        Args:
+            outputs ([`BeitForSemanticSegmentation`]):
+                Raw outputs of the model.
+            target_sizes (`torch.Tensor` of shape `(batch_size, 2)` or `List[Tuple]` of length `batch_size`, *optional*):
+                Torch Tensor (or list) corresponding to the requested final size (h, w) of each prediction. If left to
+                None, predictions will not be resized.
+        Returns:
+            semantic_segmentation: `List[torch.Tensor]` of length `batch_size`, where each item is a semantic
+            segmentation map of shape (w, h) corresponding to the target_sizes entry (if `target_sizes` is specified).
```
Comment (NielsRogge) [inline src/transformers/models/beit/feature_extraction_beit.py:241]: To me it's a bit counterintuitive to output (w, h) if we ask the target_sizes to be (h, w)

=> so I'd use the same format for both

File: src/transformers/models/beit/feature_extraction_beit.py [hunk 3]
```diff
@@ -222,3 +225,44 @@ def __call__(
...
+                )
+
+            if is_torch_tensor(target_sizes):
+                target_sizes = target_sizes.numpy()
+
+            resized_maps = []
+            semantic_segmentation = semantic_segmentation.numpy()
+
+            for idx in range(len(semantic_segmentation)):
+                resized = self.resize(image=semantic_segmentation[idx], size=target_sizes[idx])
```
Comment (NielsRogge) [inline src/transformers/models/beit/feature_extraction_beit.py:261]: As discussed offline, please use `nn.functional.interpolate` here.

```
logits = nn.functional.interpolate(outputs.logits.detach().cpu(),
                size=image.size[::-1], # (height, width)
                mode='bilinear',
                align_corners=False)
```

File: src/transformers/models/beit/feature_extraction_beit.py (hunk 2 above)
Comment (alaradirik) [reply src/transformers/models/beit/feature_extraction_beit.py]: Agreed, just fixed this

**Review by NielsRogge — APPROVED** [review]
LGTM, although it will require an update when supporting the TF model.

**Comment by LysandreJik** [comment]
Hey @alaradirik, please also ping a core maintainer for review before merging PRs.

**Pull request merged**